### PR TITLE
AF: per-channel volume loss weighting (upweight nut/p)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

AirfRANS vol_mse currently averages MSE uniformly over all 4 target channels (u_x, u_y, p, nut) before scaling by the vol-loss-weight. This uniform treatment may be suboptimal because:

1. **nut (turbulent viscosity)** is physically the hardest RANS channel to predict — it depends on turbulence closure and is typically the highest-variance field in the volume
2. **p (pressure)** dominates early training error by ~300x over ux/uy/nut (observed at PR #2755 ep201: Ux=4.97e-05, Uy=1.09e-05, nut=1.05e-05, p=0.01555)
3. Upweighting the hardest channels in the **vol loss** focuses gradient signal on reducing the channels that contribute most to vol_mse aggregation

The current vol_loss computation in `train.py` ~line 776:
```python
vol_loss = F.mse_loss(outputs["volume_preds"][batch.volume_mask], target[batch.volume_mask])
```
This is a uniform `reduction='mean'` over all N_vol_nodes × 4 values. Per-channel weighting requires computing per-channel MSE first, then taking a weighted sum.

## Target

- **vol_mse < 0.0017** (SpiderSolver external target)
- **surface_mse ≤ 0.000296** (hard constraint — do not regress surface)

## Current Baseline

| Metric | Value | Source |
|--------|-------|--------|
| surface_mse | 0.000296 | PR #3135, W&B: sh2zyfwr |
| vol_mse | 0.002039 | PR #3135, W&B: sh2zyfwr |

## Implementation

Replace the uniform `F.mse_loss` call in the AirfRANS volume loss block (train.py ~line 776) with a per-channel weighted MSE. Add a new CLI argument `--vol-channel-weights` (4 floats for [ux, uy, p, nut] channels).

Implementation pattern:
```python
# Instead of: vol_loss = F.mse_loss(preds, target)
preds = outputs["volume_preds"][batch.volume_mask]   # shape [N_vol, 4]
tgt   = transform.apply(batch.volume_y)[batch.volume_mask]  # shape [N_vol, 4]

if vol_channel_weights is not None:
    ch_w = torch.tensor(vol_channel_weights, device=preds.device, dtype=preds.dtype)
    per_ch = ((preds - tgt) ** 2).mean(dim=0)       # shape [4]
    vol_loss = (per_ch * ch_w).sum() / ch_w.sum()   # normalized weighted avg
else:
    vol_loss = F.mse_loss(preds, tgt)                # fallback to baseline
```

The `ch_w.sum()` normalization ensures the total gradient magnitude stays comparable to the baseline (important: do NOT skip this or effective vol-loss-weight changes).

## Trials

Run both trials with the standard champion config. Use `--wandb_group af-per-channel-vol-weight` to group runs in W&B.

**Trial T1 — nut-heavy** (upweight turbulent viscosity only):
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 \
  --grad-clip 1.0 --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 \
  --vol-loss-weight 10.0 \
  --vol-channel-weights 1 1 1 4 \
  --wandb_group af-per-channel-vol-weight \
  --wandb_name af-vol-ch-T1-nut4x
```
Channel weights [ux=1, uy=1, p=1, nut=4] — nut weighted 4x relative to ux/uy/p.

**Trial T2 — p+nut heavy** (upweight both pressure and turbulent viscosity):
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 \
  --grad-clip 1.0 --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 \
  --vol-loss-weight 10.0 \
  --vol-channel-weights 1 1 2 4 \
  --wandb_group af-per-channel-vol-weight \
  --wandb_name af-vol-ch-T2-p2x-nut4x
```
Channel weights [ux=1, uy=1, p=2, nut=4] — pressure 2x, nut 4x.

## Reporting

After both trials complete, report:
- `val_primary/surface_mse` and `val_primary/volume_mse` at best checkpoint (selected on joint `surface_mse + vol_mse`)
- `test_primary/surface_mse` and `test_primary/volume_mse` at that same checkpoint
- W&B run IDs for both trials
- Per-channel vol MSE breakdown at best checkpoint if available (ux, uy, p, nut separately)
- Which checkpoint selection criterion was used (surface-only vs joint)

**Important**: Use joint checkpoint selection (minimize `surface_mse + vol_mse`) not surface-only. Surface-anchored selection hides vol improvements (observed in PR #3204: at ep763 vol was still descending when surface plateau triggered early stopping).

## Motivation / Analysis

The key insight is that `vol_mse = mean(per_channel_MSE)` — channels that train easily (ux, uy) implicitly have equal gradient signal to channels that are hard (nut, p). By upweighting hard channels in the loss, we push the optimizer to allocate more capacity to reducing the hardest contributors to aggregate vol_mse.

Note: we are NOT changing the metric itself — vol_mse is still the unweighted mean. We are only changing what the training loss minimizes. This is standard practice in multi-task learning.

## Config anchors (do not change)
- optimizer: AdamW lr=6e-4
- cosine T_max=50
- grad-clip 1.0, weight-decay 1e-2
- EMA decay 0.999
- Fourier features enabled
- Architecture: 2L / 256d / 4H
- vol-loss-weight: 10.0 (do not change this; the per-channel weights are applied inside the vol loss computation)

### CODE REVIEW

**Comment by tcapelle** [comment]
Closing and re-creating with correct base branch. This PR was accidentally created targeting `main` instead of `radford`. The experiment instructions remain unchanged — tanjiro will get a new PR with the correct base.
